### PR TITLE
Fix errors with nested subfields

### PR DESCRIPTION
**Pull request by ccl-core**

### CODE REVIEW

**Comment by ccl-core** [comment]
Tested with:

```
python load.py --jsonld   /usr/local/google/home/stanzelc/croissant/datasets/1.0/huggingface-lmms-eval-lite/metadata.json --record_set gqa  --debug --num_records 3 
```

**Comment by ccl-core** [comment]
TODO: add tests

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py [hunk 1]
```diff
@@ -203,7 +204,9 @@ def call(self, df: pd.DataFrame) -> Iterator[dict[str, Any]]:
             for _, row in df.iterrows():
                 yield dict(row)
             return
-        fields = self._fields()
+        fields = self._get_fields(self.node.fields)
+        if not fields:
+            raise ValueError(f"RecordSet {self.node.uuid} has no fields!")
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py:209]: Is this wanted? We probably want to return an empty dict?

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py [hunk 2]
```diff
@@ -226,9 +229,13 @@ def _get_result(row):
...
-                    value = [
-                        _cast_value(self.node.ctx, v, field.data_type) for v in value
-                    ]
+                    try:
+                        value = [
+                            _cast_value(self.node.ctx, v, field.data_type)
+                            for v in value
+                        ]
+                    except TypeError:
+                        value = value
```
Comment (marcenacp) [inline python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py:238]: What do you think of dropping the _cast_value?

**Review by marcenacp — COMMENTED** [review]
Thanks for the thorough bug analysis!

File: python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py (hunk 2 above)
Comment (ccl-core) [reply python/mlcroissant/mlcroissant/_src/operation_graph/operations/field.py:238]: Are you sure we want to do that for all cases?

**Pull request merged**